### PR TITLE
BatchAI. Several fixes

### PR DESCRIPTION
**Pull request by AlexanderYukhanov**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json [hunk 1]
```diff
@@ -3774,8 +3774,18 @@
       },
       "description": "Values returned by the List operation."
     },
+    "ExperimentBaseProperties": {
+      "properties": {
```
[the remark this answers is not in the document]
Comment (AlexanderYukhanov) [reply specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json]: If we do not specify it, autogenerated C# client still has ExperimentCreationParameter which has type Object. Object will be very confusing for users.

**Review by ravbhatnagar — COMMENTED** [review]
ARM does not need to sign off on this change. Anyway, this is something done for better generated code. removing the ARM signoff label.

File: specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json]: @AlexanderYukhanov  I mean, what's the purpose of the type?

[the remark this answers is not in the document]
Comment (fearthecowboy) [reply specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json]: Ok, I'm missing something here. 

`ExperimentBaseProperties` doesn't contain *anything*

`ExperimentCretationParameters` only has a reference to `ExperimentBaseProperties` and declares `x-ms-flatten` .. What exactly are you asking users to pass into this function? What good are empty objects without any definition?

[the remark this answers is not in the document]
Comment (AlexanderYukhanov) [reply specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json]: We just need to have an experiment create api which doesn't require any parameters but without this helper structure the generated code looks like this:

        public async Task<AzureOperationResponse<Experiment>> CreateWithHttpMessagesAsync(string resourceGroupName, string workspaceName, string experimentName, object parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))

Please notice "oject parameters" in the signature. 
If we add this structure the parameters have reasonable type ExperimentCreateParameters and become optional.

[the remark this answers is not in the document]
Comment (AlexanderYukhanov) [reply specification/batchai/resource-manager/Microsoft.BatchAI/stable/2018-05-01/BatchAI.json]: We have removed ExperimentCreationParameters completely now.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**